File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.size();
        int len=0;
        map<char,int>mp;
        int i=0;
        for(int j=0;j<n;j++){
            if(mp.find(s[j])!=mp.end()){
                i=max(mp[s[j]]+1,i);
            }
            
            len=max(j-i+1,len);
            mp[s[j]]=j;
        }
        return len;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i,n,maxi=0,len(s),0
        mapp=dict()
        for j in range(n):
            if s[j] in mapp:
                i=max(mapp.get(s[j],0)+1,i)
            maxi=max(maxi,j-i+1)
            mapp[s[j]]=j
        return maxi
            

      